app/src/main/java/com/babylon/wallet/android/domain/usecases/assets/ObserveResourceUseCase.kt

```diff
@@ -0,0 +1,36 @@
+package com.babylon.wallet.android.domain.usecases.assets
+
+import com.babylon.wallet.android.data.repository.state.StateRepository
+import com.babylon.wallet.android.domain.model.resources.Resource
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import javax.inject.Inject
+
+/**
+ * Fetches details regarding this fungible/non-fungible by looking it up through its address.
+ * If an account address is provided, the repository will try to also fetch amount information.
+ */
+class ObserveResourceUseCase @Inject constructor(
```

Comment:
Hey @jakub-rdx it is indeed unused since I created this to test how the resources details are being fetched when used inside the dynamic bottom sheets. It is actually used in the next PR.

> this is a bit hidden knowledge that when collecting from this flow we should also handle errors.

Isn't that the case for all flows? We have the `catch` operator built in. Do you need me to mention it in the comments?